### PR TITLE
config: Revise SimpleLRUCache Configuration 

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/17567

### What is changed and how it works?

1. introduce new configuration to control the mem quota of a single tidb-server
2. deprecate `max-memory` configuration
3. set memory quota for plancache as `server-mem-quota` instead of `max-memory`

### Release note 
- Add configuration `server-memory-quota` control the tidb-server memory quota
- Make `max-memory` deprecated

### CODE REVIEW

File: config/config.toml.example [hunk 1]
```diff
@@ -198,8 +198,12 @@ record-db-qps = false
 max-procs = 0
 
 # Max memory size to use, 0 use the total usable memory in the machine.
+# The maximum memory limit for the Prepared Least Recently Used (LRU) caching
```
Comment (SunRunAway) [inline config/config.toml.example:201]: Could you make this change in a separate PR because it's needed to cherry-pick to v2.1, v3.0, v3.1, and v4.0.

Comment (Yisaer) [reply config/config.toml.example:201]: sure

File: config/config.toml.example [hunk 2]
```diff
@@ -200,6 +200,9 @@ max-procs = 0
 # Max memory size to use, 0 use the total usable memory in the machine.
 max-memory = 0
 
+## Memory size quota for tidb server, 0 means unlimited
```
Comment (SunRunAway) [inline config/config.toml.example:203]:
```suggestion
# Memory size quota for tidb server, 0 means unlimited
```

**Comment by SunRunAway** [comment]
The PR title has to be changed.

**Comment by Yisaer** [comment]
@SunRunAway  PTAL when you are free.
